src/auction.js

```diff
@@ -0,0 +1,433 @@
+import { uniques, timestamp } from './utils';
+import { getPriceBucketString } from './cpmBucketManager';
+import { nativeBidIsValid } from './native';
+import { store } from './videoCache';
+import { Renderer } from 'src/Renderer';
+import { config } from 'src/config';
+
+const utils = require('./utils');
+const adaptermanager = require('./adaptermanager');
+// To be injected as dependency once #1453 is merged
+const events = require('./events');
+const CONSTANTS = require('./constants.json');
+
+const AUCTION_STARTED = 'started';
+const AUCTION_IN_PROGRESS = 'inProgress';
+const AUCTION_COMPLETED = 'completed';
+
+// register event for bid adjustment
+events.on(CONSTANTS.EVENTS.BID_ADJUSTMENT, function (bid) {
+  adjustBids(bid);
+});
+
+function Auction(customPriceBucket, granularity) {
+  var _id = utils.getUniqueIdentifierStr();
+  var _adUnits = [];
+  var _adUnitCodes = [];
+  var _targeting = [];
+  var _bidderRequests = [];
+  var _bidsReceived = [];
+  var _auctionStart;
+  var _auctionId = utils.getUniqueIdentifierStr();
+  var _auctionStatus;
+  var _callback;
+  var _timer;
+
+  this.setAdUnits = (adUnits) => _adUnits = adUnits;
```

Comment:
Whenever possible, it's a good idea to minimize mutability.

For example, the only use of `setAdUnits` that I see is in `requestBids`:

```
  const auction = auctionManager.createAuction();
  auction.setAdUnits(adUnits);
  auction.setAdUnitCodes(adUnitCodes);
```

This could easily be replaced by something like:

```
const auction = auctionManager.createAuction({
  adUnits: adUnits,
  adUnitCodes: adUnitCodes
});
```

Doing so lets you delete the `set` function, which makes your code easier to reason about.